src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -126,6 +127,9 @@ public Test BuildFrom(IMethodInfo method, Test parentSuite)
                     needCombinatorial = false;
             }
 
+            // This check must be done before CombinatorialAttribute gets added to the builders collection
+            var shouldForceParameterization = builders.Count > 0;
```

Comment:
My instinct would be to check wither the test method has parameters. If not, return a single, If so, return a suite. But you might know of a scenario where that won't work.